planner/core/find_best_task.go

```diff
@@ -248,6 +248,13 @@ func (p *baseLogicalPlan) enumeratePhysicalPlans4Task(physicalPlans []PhysicalPl
 				continue
 			}
 		}
+		// convert to root task as needed to satisfy property
+		if prop.TaskTp == property.RootTaskType {
+			_, ok := curTask.(*rootTask)
+			if prop.IsFlashProp() || !ok {
```

Comment:
If prop's task type is root task, It can never be a flash prop

Reply:
I did it as refered to your PR of [line](https://github.com/pingcap/tidb/pull/20894/files#diff-f603dfc91719f7335c1d0ee9a59824d0d47ad43946271ea798827c7c3564c0b2R246)

updated.